go/weed/weed_server/filer_server_handlers.go

```diff
@@ -134,14 +161,63 @@ func (fs *FilerServer) PostHandler(w http.ResponseWriter, r *http.Request) {
 	if collection == "" {
 		collection = fs.collection
 	}
-	assignResult, ae := operation.Assign(fs.master, 1, replication, collection, query.Get("ttl"))
-	if ae != nil {
-		glog.V(0).Infoln("failing to assign a file id", ae.Error())
-		writeJsonError(w, r, http.StatusInternalServerError, ae)
+
+	buf, _ := ioutil.ReadAll(r.Body)
+	r.Body = analogueReader{bytes.NewBuffer(buf)}
+	fileName, _, _, _, _, _, _, pe := storage.ParseUpload(r)
+	if pe != nil {
+		glog.V(0).Infoln("failing to parse post body", pe.Error())
+		writeJsonError(w, r, http.StatusInternalServerError, pe)
+		return
+	}
+	//reconstruct http request body for following new request to volume server
+	r.Body = analogueReader{bytes.NewBuffer(buf)}
+
+	path := r.URL.Path
+	if strings.HasSuffix(path, "/") {
+		if fileName != "" {
+			path += fileName
+		}
+	}
+	var fileId string
+	var err error
+	var urlLocation string
+	if fileId, err = fs.filer.FindFile(path); err != nil {
+		glog.V(0).Infoln("failing to find path in filer store", path, pe.Error())
+		writeJsonError(w, r, http.StatusInternalServerError, err)
 		return
+	} else if fileId != "" {
+		parts := strings.Split(fileId, ",")
+		if len(parts) != 2 {
+			glog.V(1).Infoln("Invalid fileId", fileId)
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+		lookup, lookupError := operation.Lookup(fs.master, parts[0])
```

Comment:
I think what you mean is the operation.LookupFileId API, I will replace the redudant code(post and get handler) with the following:
		var le error
		urlLocation, le = operation.LookupFileId(fs.master, fileId)
		if le != nil {
			glog.V(1).Infoln("operation LookupFileId %s failed, err is %s", fileId, le.Error())
			w.WriteHeader(http.StatusNotFound)
			return
		}